Physics components (RigidBody, Collider) not auto-registered when physics enabled

## Description

When physics is enabled via `-Dphysics=true`, user projects still need to manually register physics components (RigidBody, Collider, Velocity) in their ComponentRegistry. This is error-prone and should be automatic.

## Error

```
/Users/alexandrecalvao/prj/labelle-toolkit/labelle-engine/scene/src/component.zig:139:20: error: struct 'main.Components__struct_21845' has no member named 'RigidBody'
```

## Current Behavior

Users must manually add physics components:

```zig
const physics = @import("physics");

pub const Components = engine.ComponentRegistry(struct {
    pub const RigidBody = physics.RigidBody;
    pub const Collider = physics.Collider;
    pub const Velocity = physics.Velocity;
    // ... other components
});
```

## Expected Behavior

When physics is enabled, the engine should automatically make physics components available for use in scenes without requiring manual registration.

## Possible Solutions

1. Auto-merge physics components into ComponentRegistry when physics module is available
2. Use `ComponentRegistryMulti` pattern to combine user components with physics components
3. Export physics components from the engine module when physics is enabled

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/root.zig
+++ b/root.zig
@@ -162,16 +162,16 @@ else
 /// have all engine components available without manual imports.
 pub const BuiltinComponentsWithPhysics = if (build_options.physics_enabled)
     struct {
-        // Render components
-        pub const Position = render.Position;
-        pub const Sprite = render.Sprite;
-        pub const Shape = render.Shape;
-        pub const Text = render.Text;
-        // Physics components (auto-included)
-        pub const RigidBody = physics.RigidBody;
-        pub const Collider = physics.Collider;
-        pub const Velocity = physics.Velocity;
-        pub const Touching = physics.Touching;
+        // Render components (from BuiltinComponents)
+        pub const Position = BuiltinComponents.Position;
+        pub const Sprite = BuiltinComponents.Sprite;
+        pub const Shape = BuiltinComponents.Shape;
+        pub const Text = BuiltinComponents.Text;
+        // Physics components (from PhysicsComponents)
+        pub const RigidBody = PhysicsComponents.RigidBody;
+        pub const Collider = PhysicsComponents.Collider;
+        pub const Velocity = PhysicsComponents.Velocity;
+        pub const Touching = PhysicsComponents.Touching;
     }
 else
     BuiltinComponents;

